src/main/java/org/spongepowered/api/GameRegistry.java

```diff
@@ -536,4 +543,110 @@
      */
     List<BannerPatternShape> getBannerPatternShapes();
 
+    /**
+     * Gets the {@link TextColor} with the provided name.
+     *
+     * @param name The name of the text color
+     * @return The {@link TextColor} with the given name or Optional.absent() if not found
+     */
+    Optional<TextColor> getTextColor(String name);
+
+    /**
+     * Gets a {@link List} of all possible {@link TextColor}s.
+     *
+     * @return The list of all available {@link TextColor}s
+     */
+    List<TextColor> getTextColors();
+
+    /**
+     * Gets the {@link TextStyle} with the provided name.
+     *
+     * @param name The name of the text style
+     * @return The {@link TextStyle} with the given name or Optional.absent() if not found
+     */
+    Optional<TextStyle> getTextStyle(String name);
+
+    /**
+     * Gets a {@link List} of all possible {@link TextStyle}s.
+     *
+     * @return The list of all available {@link TextStyle}s
+     */
+    List<TextStyle> getTextStyles();
+
+    /**
+     * Gets the {@link ChatType} with the provided name.
+     *
+     * @param name The name of the chat type
+     * @return The {@link ChatType} with the given name or Optional.absent() if not found
+     */
+    Optional<ChatType> getChatType(String name);
+
+    /**
+     * Gets a {@link List} of all possible {@link TextStyle}s.
+     *
+     * @return The list of all available {@link TextStyle}s
+     */
+    List<ChatType> getChatTypes();
+
+    /**
+     * Gets the {@link SelectorType} with the provided name.
+     *
+     * @param name The name of the selector type
+     * @return The {@link SelectorType} with the given name or Optional.absent() if not found
+     */
+    Optional<SelectorType> getSelectorType(String name);
+
+    /**
+     * Gets a {@link List} of all possible {@link ArgumentType}s.
+     *
+     * @return The list of all available {@link ArgumentType}s
+     */
+    List<ArgumentType<?>> getArgumentTypes();
+
+    /**
+     * Gets the {@link ArgumentType} with the provided name.
+     *
+     * @param name The name of the argument type
+     * @return The {@link ArgumentType} with the given name or Optional.absent() if not found
+     */
+    Optional<ArgumentType<?>> getArgumentType(String name);
```

Comment:
The method order is reversed here in comparison to the rest of the GameRegistry.

Everywhere else there is first the `Stuff getStuff(By param)` and then the `List getStuff()` method.